pipeline/stages/deploy.yaml

```diff
@@ -40,8 +40,64 @@ stages:
         - template: ../variables/variables-${{parameters.env}}.yaml
         - group: cvp-${{parameters.env}}
       jobs:
+        - job: VmOn${{parameters.env}}
+          displayName: 'Turn on VMs'
+          steps:
+          - task: AzureCLI@2
+            name: 'VmOn${{parameters.env}}'
+            displayName: 'Turning Vm On ${{parameters.env}}'
+            condition: always()
+            inputs:
+              azureSubscription: '${{variables.subscriptionName}}'
+              scriptType: pscore
+              scriptLocation: inlineScript
+              inlineScript: |
+                $azdo_pipe_to_change_vm_status="$env:AZDO_PIPE_TO_CHANGE_VM_STATUS"
+                $vm_resting_state_on="$env:VM_RESTING_STATE_ON"
+                $resourcegroup="cvp-recordings-${{parameters.env}}-rg"
+
+                Write-Output "Should state be altered by azdo pipeline: $azdo_pipe_to_change_vm_status"
+                Write-Output "Should resting state be on: $vm_resting_state_on"
+                Write-Output ""
+
+                $rgExists = az group exists --name $resourcegroup
+                if ( $azdo_pipe_to_change_vm_status )
```

Comment:
you're right about the 'if' clause. i don't think it's necessary any longer. 

It was originally put in prior to the additional of the environment variables. So $azdo_pipe_to_change_vm_status is redundant.

what it was originally doing was checking whether the pipeline should have been allowed to change the status.

if you're referecning the '$rgExists = az group exists --name $resourcegroup' piece. if the az cli throws an error, this can't be caught be the try catch. it's safer then to check for the existence of the rg and then if present, continue.

Reply:
Removed.